test/unit/util/mapbox.test.js

```diff
@@ -322,82 +304,219 @@ test("mapbox", (t) => {
+                callback();
+            });
 
-            const req = window.server.requests[0];
-            t.false(req);
+            t.test('does not POST event when previously stored data is on the same day', (t) => {
+                config.ACCESS_TOKEN = 'pk.*';
+                const now = +Date.now();
 
-            t.end();
-        });
+                window.localStorage.setItem(`mapbox.turnstileEventData:${config.ACCESS_TOKEN}`, JSON.stringify({
+                    anonId: uuid(),
+                    lastSuccess: now
+                }));
 
-        t.test('POSTs appuserTurnstile event when access token changes', (t) => {
-            config.ACCESS_TOKEN = 'pk.new.*';
+                t.stub(browser, 'now').callsFake(() => now + 5); // A bit later
 
-            mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+                mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
 
-            const req = window.server.requests[0];
-            req.respond(200);
+                t.false(window.server.requests.length);
+                t.end();
+            });
 
-            const reqBody = JSON.parse(req.requestBody)[0];
-            t.equal(req.url, `${config.EVENTS_URL}?access_token=pk.new.*`);
-            t.equal(req.method, 'POST');
-            t.equal(reqBody.event, 'appUserTurnstile');
-            t.equal(reqBody.sdkVersion, version);
-            t.ok(reqBody.userId);
+            t.test('POSTs event when previously stored anonId is not a valid uuid', (t) => {
+                config.ACCESS_TOKEN = 'pk.*';
+                const now = +Date.now();
 
-            t.end();
-        });
+                window.localStorage.setItem(`mapbox.turnstileEventData:${config.ACCESS_TOKEN}`, JSON.stringify({
+                    anonId: 'anonymous',
+                    lastSuccess: now
+                }));
 
-        t.test('POSTs appuserTurnstile event on next calendar day', (t) => {
-            config.ACCESS_TOKEN = 'pk.*';
-            const today = Date.now();
-            const tomorrow = today + (25 * 60 * 60 * 1000); // Add a day
-            t.stub(browser, 'now').callsFake(() => tomorrow);
+                t.stub(browser, 'now').callsFake(() => now + ms25Hours); // next day
+
+                mapbox.postTurnstileEvent(['a.tiles.mapbox.com']);
+
+                const req = window.server.requests[0];
+                req.respond(200);
+
+                const reqBody = JSON.parse(req.requestBody)[0];
+                t.notEqual(reqBody.userId, 'anonymous');
+                t.end();
+            });
+
+            t.test('POSTs event when previously stored timestamp is in the future', (t) => {
+                config.ACCESS_TOKEN = 'pk.*';
+                const now = +Date.now();
+
+                window.localStorage.setItem(`mapbox.turnstileEventData:${config.ACCESS_TOKEN}`, JSON.stringify({
+                    anonId: uuid(),
+                    lastSuccess: now + ms25Hours // 24-hours later
+                }));
```

Comment:
I think it would be better to test this by calling `postTurnstileEvent` twice, with the (stubbed) time being set back between calls, rather than setting `localStorage` directly

Reply:
Are you implying that the test should cover the case of the queue receiving events in an out-of-order time sequence? In a single threaded environment, that situation seems highly unlikely. Even if it does happen, it should be counted as a single session IMHO. 

Testing with `LocalStorage` allows verifying that clocks going back in time between sessions are accounted for, where it should be posted as a separate session correctly.

Reply:
Oh -- no, I just meant that if possible, we should do this test using calls to `postTurnstileEvent` (the api being tested) rather than by directly manipulating `localStorage` (which requires baking more implementation details, like the storage key, into the test)

Reply:
> we should do this test using calls to postTurnstileEvent

I believe that within a single session, the clock going backwards should not report a new event -  so this cannot be tested with calls to the method directly. To simulate calls across sessions, the best way is to change the access token and clear out the cached `eventData`

Reply:
> this cannot be tested with calls to the method directly

Why not, with something like:

```js
let now = Date.now();
t.stub(browser, 'now').callsFake(() => now);

mapbox.postTurnstileEvent();

// simulate system clock being set backwards
now = now - 1000 * 60 * 60 * 24;

mapbox.postTurnstileEvent();
```

This is a more direct simulation of the situation we're intending to test, and allows the test to avoid a dependency on implementation details.

Reply:
That makes sense to add as a negative test, `when clock moves bacward, new event is not set.`